src/test/java/org/cactoos/map/SolidMapTest.java

```diff
@@ -81,4 +86,71 @@ public void mapsToSameObjects() throws Exception {
             map.get(0), Matchers.equalTo(map.get(0))
         );
     }
+
+    @Test
+    @SuppressWarnings({"unchecked", "MismatchedQueryAndUpdateOfCollection"})
+    public void extendsExistingMapWithArrayOfEntries() {
+        final SolidMap<Integer, Integer> map = new SolidMap<>(
+            new MapEntry<>(0, 10),
+            new MapEntry<>(1, 11)
+        );
+        final SolidMap<Integer, Integer> toadd = new SolidMap<>(
+            new MapEntry<>(2, 12),
+            new MapEntry<>(3, 13)
+        );
+        final Map.Entry<Integer, Integer>[] array =
+            (Map.Entry<Integer, Integer>[]) new Map.Entry<?, ?>[toadd.size()];
+        toadd.entrySet().toArray(array);
+        MatcherAssert.assertThat(
+            "Does not extends existing map with array of entries",
+            new SolidMap<Integer, Integer>(map, array).size(),
+            Matchers.equalTo(map.size() + toadd.size())
+        );
+    }
+
+    @Test
+    public void mapsIterableWithKeyFuncAndValueFunc() {
+        MatcherAssert.assertThat(
+            "Functions are not applied to key and value",
+            new SolidMap<>(
+                key -> new SubText(new TextOf(key), 0, 1).asString(),
+                value -> new UpperText(new TextOf(value)).asString(),
+                new IterableOf<>("aa", "bb")
+            ),
+            Matchers.allOf(
```

Comment:
@flslkxtc Can you also add a matcher for the size? To check that there are only 2 entries?